### PR TITLE
Add client test for student-exam-details

### PR DESCRIPTION
**Pull request by kriststoja**
### Checklist
- [x] Client: I followed the [coding and design guidelines](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/client.html).
- [x] Client: I added integration tests (Jest) related to the features (with a high test coverage)

### Motivation and Context
Improving test coverage

### Description
Client test for student-exam-details-component

### Test Coverage
student-exam-details.component.ts 93.1%

### CODE REVIEW

File: src/test/javascript/spec/component/exam/manage/student-exams/student-exam-detail.component.spec.ts [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+
+    beforeEach(() => {
+        course = new Course();
+        course.id = 1;
+
+        student = new User();
+        student.name = 'name';
+        student.login = 'login';
+        student.email = 'email';
+        student.visibleRegistrationNumber = 'visibleRegistrationNumber';
```
Comment (ivanchimeno) [inline src/test/javascript/spec/component/exam/manage/student-exams/student-exam-detail.component.spec.ts:62]: Same as above, you can save some extra lines by doing:
```
student: Student = { name: "name", login:"login", ... }
```

File: src/test/javascript/spec/component/exam/manage/student-exams/student-exam-detail.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+
+        exercise = new QuizExercise(course, exerciseGroup);
+        expect(studentExamDetailComponent.exerciseIcon(exercise)).to.equal('check-double');
+
+        exercise = new FileUploadExercise(course, exerciseGroup);
+        expect(studentExamDetailComponent.exerciseIcon(exercise)).to.equal('file-upload');
+    });
+
+    it('should save working time', () => {
+        const studentExamService = TestBed.inject(StudentExamService);
```
Comment (ivanchimeno) [inline src/test/javascript/spec/component/exam/manage/student-exams/student-exam-detail.component.spec.ts:190]: I'm not sure but is it best practice to move this line into `beforeEach`?

File: src/test/javascript/spec/component/exam/manage/student-exams/student-exam-detail.component.spec.ts [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+                studentExamDetailComponentFixture = TestBed.createComponent(StudentExamDetailComponent);
+                studentExamDetailComponent = studentExamDetailComponentFixture.componentInstance;
+            });
+    });
+    afterEach(() => {
+        sinon.restore();
+    });
+
+    it('initialize', () => {
+        const courseManagementService = TestBed.inject(CourseManagementService);
```
Comment (ivanchimeno) [inline src/test/javascript/spec/component/exam/manage/student-exams/student-exam-detail.component.spec.ts:164]: I'm not sure but is it best practice to move this line into beforeEach?

**Review by ivanchimeno — CHANGES_REQUESTED** [review]
The tests look good, I just added couple optional remarks

File: src/test/javascript/spec/component/exam/manage/student-exams/student-exam-detail.component.spec.ts [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+    let student: User;
+    let studentExam: StudentExam;
+    let exercise: Exercise;
+    let exerciseGroup: ExerciseGroup;
+    let exam: Exam;
+    let studentParticipation: StudentParticipation;
+    let result: Result;
+
+    beforeEach(() => {
+        course = new Course();
```
[the remark this answers is not in the document]
Comment (kriststoja) [reply src/test/javascript/spec/component/exam/manage/student-exams/student-exam-detail.component.spec.ts]: Looks better with your suggestion. Done

File: src/test/javascript/spec/component/exam/manage/student-exams/student-exam-detail.component.spec.ts (hunk 1 above)
Comment (kriststoja) [reply src/test/javascript/spec/component/exam/manage/student-exams/student-exam-detail.component.spec.ts]: Done

File: src/test/javascript/spec/component/exam/manage/student-exams/student-exam-detail.component.spec.ts (hunk 2 above)
Comment (kriststoja) [reply src/test/javascript/spec/component/exam/manage/student-exams/student-exam-detail.component.spec.ts]: I think you right, it is better. Did the changes

**Review by nicolasruscher — COMMENTED** [review]
Code looks good after the changes, could you add the new test coverage percentages for this component? 👍

**Comment by kriststoja** [comment]
> Code looks good after the changes, could you add the new test coverage percentages for this component? 👍

93.1%

**Review by krusche — APPROVED** [review]

**Pull request merged**